pkg/client/restclient/config.go

```diff
@@ -45,11 +45,11 @@ const (
 // Config holds the common attributes that can be passed to a Kubernetes client on
 // initialization.
 type Config struct {
-	// Host must be a host string, a host:port pair, or a URL to the base of the apiserver.
-	// If a URL is given then the (optional) Path of that URL represents a prefix that must
-	// be appended to all request URIs used to access the apiserver. This allows a frontend
-	// proxy to easily relocate all of the apiserver endpoints.
-	Host string
+	// Hosts must be a slice of host strings, host:port pairs, or a URLs to the base of the
+	// apiservers. If a URL is given then the (optional) Path of that URL represents a prefix
+	// that must be appended to all request URIs used to access the apiserver. This allows a
+	// frontend proxy to easily relocate all of the apiserver endpoints.
+	Hosts []string
```

Comment:
Rather than change this, I would prefer:

```
Host string
AlternateHosts []string
```

That reduces the scope of your PR here significantly and confines alternate hosts to the use case it was intended for.  A user must always specify at least one host, but alternates are checked afterwards.